src/components/Input/Input.component.tsx

```diff
@@ -0,0 +1,83 @@
+/* eslint-disable @typescript-eslint/indent */
+import React from 'react';
+
+import {Icon} from '@Components/Icon';
+import {extendClassname} from '@Utils';
+
+import {StyledInput, StyledFieldSet, StyledInputContainer} from './Input.styles';
+import {InputProps} from './Input';
+
+/**
+ * DisplayFieldSet - sub-component
+ *
+ * Displays fieldset for input component
+ */
+const DisplayFieldSet = ({value, error, label}: InputProps) => {
+    return (
+        <StyledFieldSet
+            className={`input_label ${extendClassname({
+                'visible_label-border': Boolean(value),
+                input_error: Boolean(error),
+            })}`
+            }
+            error={error}
+        >
+            <legend className={extendClassname({visible_label: Boolean(value)})}>
+                {label}
+            </legend>
+        </StyledFieldSet>
+    );
+};
+
+/**
+ * Input Component
+ *
+ * @example Correct usage
+ * ```ts
+ * <Input label='label'/>
+ * ```
+ *
+ * @example Correct usage with icon
+ * ```ts
+ * <Input label='label' iconName='Globe'/>
+ * ```
+ */
+export const Input = (props: InputProps) => {
+    const {label, iconName, className = '', fullWidth, ...restProps} = props;
+    const {readOnly, error, required, value = ''} = restProps;
+
+    const inputLabel = required ? `${label} *` : label;
```

Comment:
```suggestion
    const inputLabel = `${label}${required && ' *' }`;
```

Reply:
if `required` is undefined then it shows label like this
![image](https://user-images.githubusercontent.com/102018792/198194849-cfb5eeae-f670-4a8e-a413-05604c4b23ff.png)